lib/v2/xmnn/news.js

```diff
@@ -0,0 +1,69 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const { category = 'xmxw' } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 30;
+
+    const rootUrl = 'https://news.xmnn.cn';
+    const currentUrl = new URL(category, rootUrl).href;
```

Comment:
Use the redirected url
https://github.com/DIYgod/RSSHub/actions/runs/6959820710/job/18937928456#step:12:18
```suggestion
    const currentUrl = new URL(`${category}/`, rootUrl).href;
```